docs/user/agent/authentication-authorization.md

```diff
@@ -0,0 +1,124 @@
+
+### Types of Tokens:
+1. **agentAuth**:
+   - **Access Level**: Read-Write (RW)
+   - **Used By**: The agent service that runs on hosts booted with the agent-based installer.
+   
+2. **userAuth**:
+   - **Access Level**: Read-Write (RW)
+   - **Used By**: Systemd services and the agent-based installer client in the Assisted Service.
+   
+3. **watcherAuth**:
+   - **Access Level**: Read-Only (RO)
+   - **Used By**: Only the `wait-for` command (on Day 1) and `monitor-add-nodes` command (on Day 2).
+
+### Token Expiry Behavior
+- **Day 1** Tokens: The tokens generated on Day 1 do not expire, meaning the agent ISO remains valid indefinitely. This is useful for use cases where customers create an agent ISO on one day and boot it on a later day (e.g., in a data center).
+- **Day 2** Tokens: Tokens generated for nodes ISO on Day 2 expire after 48 hours. This ensures that the nodes.iso is only valid for a short period.
+
+### Token Management
+When a Day 2 node ISO is created, all three tokens are saved in the `openshift-config` namespace as a new secret named `agent-auth-token`. The following steps outline how token management works:
+1. If the secret exists in the cluster, the token can be retrieved.
+2. All tokens expire at the same time, so only one token needs to be checked for expiry.
+3. The expiration time is calculated as 24 hours before the token's expiry time.
+4. If the token in the secret is older than 24 hours when creating a node ISO, the installer generates new tokens. These new tokens are saved in the asset store, and the secret in the cluster is updated with the new token values.
+5. If the token in the secret is not older than 24 hours, the installer retrieves the existing token from the cluster, updates the asset store with the retrieved token, and uses the retrieved token for the new node ISO.
+
+## Authorization
+
+To implement authorization for the Agent-Based Installer, a new security definition is added in the `swagger.yaml` file in Assisted Service. The new security definition is as follows:
+
+```yaml
+watcherAuth:
+  type: apiKey
+  in: header
+  name: Watcher-Authorization
+```
+
+The required endpoints are annotated with this security definition. For example:
+
+```yaml
+/v2/clusters:
+  get:
+    tags:
+      - installer
+    security:
+      - userAuth: [admin, read-only-admin, user]
+      - watcherAuth: []
+    description: Retrieves the list of OpenShift clusters.
+    operationId: v2ListClusters
+```
+
+### Authorization Handler
+
+The Assisted Service API includes an authorization handler implemented in the `pkg/auth/agent_local_authz_handler.go` file. The `AgentLocalAuthzHandler`  is responsible for verifying that the `auth_scheme` in the token matches the required `authScheme` for the endpoint. The authorization logic works as follows:
```

Comment:
A direct link to the real code could be useful here

Reply:
Done